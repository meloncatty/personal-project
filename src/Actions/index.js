import {apiKey} from '../apiKey.js'

export const resultsSuccess = (results) => ({
  type: 'QUERY_RESULTS_SUCCESS',
  results
})

export const resultsAreLoading = (bool) => ({
  type: 'RESULTS_ARE_LOADING',
  resultsLoading: bool
})

export const resultsHaveErrored = (bool) => ({
  type: 'RESULTS_HAVE_ERRORED',
  resultsErrored: bool
})

export const fullArticleLoading = (bool) => ({
  type: 'FULL_ARTICLE_LOADING',
  articleLoading: bool
})

export const fullArticleErrored = (bool) => ({
  type: 'FULL_ARTICLE_ERRORED',
  articleErrored: bool
})

export const fullArticleSuccess = (result) => ({
  type: 'FULL_ARTICLE_SUCCESS',
  result
})

export const userAuthentication = (userData) => ({
  type: 'USER_AUTHENTICATION',
  userAuth: userData
})

export const isUserSignedIn = (bool) => ({
  type: 'IS_USER_SIGNED_IN',
  isUserSignedIn: bool
})

export const captureUserArticles = (articles) => ({
  type: 'CAPTURE_USER_ARTICLES',
  userArticles: articles
})

export const fetchArticles = (query) => {
  return async (dispatch) => {
    try {
      dispatch(resultsSuccess([]))
      dispatch(resultsHaveErrored(false))
      dispatch(resultsAreLoading(true))
      const url = `https://core.ac.uk:443/api-v2/articles/search/${query}?page=1&pageSize=10&metadata=true&fulltext=false&citations=true&similar=false&duplicate=false&urls=true&faithfulMetadata=false&apiKey=${apiKey}`
      const response = await fetch(url)
      if(!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(resultsAreLoading(false))
      const articles = await response.json()
      dispatch(resultsSuccess(articles.data))
    } catch (error) {
      dispatch(resultsHaveErrored(true))
      dispatch(resultsAreLoading(false))
    }
  }
}

export const fetchFullText = (id) => {
  return async (dispatch) => {
    try {
      dispatch(fullArticleLoading(true))
      const url = `https://core.ac.uk:443/api-v2/articles/get/${id}?metadata=true&fulltext=true&citations=false&similar=false&duplicate=false&urls=false&faithfulMetadata=false&apiKey=${apiKey}`
      const response = await fetch(url)
      if(!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(fullArticleLoading(false))
      const article = await response.json()
      dispatch(fullArticleSuccess(article.data))
    } catch (error) {
      dispatch(fullArticleErrored(true))
      dispatch(fullArticleLoading(false))
    }
  }
}
