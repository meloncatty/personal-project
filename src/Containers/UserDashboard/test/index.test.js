import React from 'react'
import {shallow, mount} from 'enzyme'
import {UserDashboard, mapDispatchToProps} from '../'
import {fetchUserArticles} from '../../../Actions'

describe('UserDashboard', () => {
  it('should match snapshot', () => {
    const mockSuccess = [{
      id: 1,
      title: 'title',
      description: 'description'
    }]
    const mockUser = ['9f9sv90b9z']
    const expected = {
      storeArticles: []
    }
    const wrapper = shallow(<UserDashboard isUserSignedIn={true} fetchUserArticles={jest.fn()} fetchUserArticleSuccess={mockSuccess} userAuthentication={mockUser} />)

    expect(wrapper).toMatchSnapshot()
  });

  it('should have default state', () => {
    const mockSuccess = [{
      id: 1,
      title: 'title',
      description: 'description'
    }]
    const mockUser = ['9f9sv90b9z']
    const expected = {
      storeArticles: []
    }
    const wrapper = shallow(<UserDashboard isUserSignedIn={true} fetchUserArticles={jest.fn()} fetchUserArticleSuccess={mockSuccess} userAuthentication={mockUser} />)

    expect(wrapper.state()).toEqual(expected)
  });

  it('should call fetchArticlesFromFirebase when mounted', () => {
    const mockSuccess = [{
      id: 1,
      title: 'title',
      description: 'description'
    }]
    const mockUser = ['9f9sv90b9z']
    const expected = {
      storeArticles: []
    }
    const wrapper = shallow(<UserDashboard isUserSignedIn={true} fetchUserArticles={jest.fn()} fetchUserArticleSuccess={mockSuccess} userAuthentication={mockUser} />)
    
    expect(wrapper.instance().fetchArticlesFromFirebase).toHaveBeenCalled
  });

  describe('fetchArticlesFromFirebase', () => {
    it('should call fetchUserArticles', () => {
      const mockSuccess = [{
        id: 1,
        title: 'title',
        description: 'description'
      }]
      const mockUser = ['L8Ko3dfZSeRVpAQPKUJJHUiF78C3']
      const expected = {
        storeArticles: []
      }
      const wrapper = shallow(<UserDashboard isUserSignedIn={true} fetchUserArticles={jest.fn()} fetchUserArticleSuccess={mockSuccess} userAuthentication={mockUser} />)
      
      expect(wrapper.instance().fetchUserArticles).toHaveBeenCalled
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch with full article id', () => {
      const mockSuccess = [{
        id: 1,
        title: 'title',
        description: 'description'
      }]
      const mockUser = ['L8Ko3dfZ8C3']
      const expected = {
        storeArticles: []
      }
      const wrapper = shallow(<UserDashboard isUserSignedIn={true} fetchUserArticles={jest.fn()} fetchUserArticleSuccess={mockSuccess} userAuthentication={mockUser} />)
      const mockDispatch = jest.fn()
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.fetchUserArticles(82413678)
      
      expect(mockDispatch).toHaveBeenCalled
    });    
  });
});