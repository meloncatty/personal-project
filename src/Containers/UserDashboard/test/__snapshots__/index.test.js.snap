// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDashboard should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserDashboard
    fetchUserArticles={[Function]}
    fetchUserArticlesSuccess={
        Array [
            Object {
              "authors": Array [
                "Low, Zhen Lin",
              ],
              "datePublished": "2016-01-06",
              "description": "We show that every small model category that satisfies certain size
        conditions can be completed to yield a combinatorial model category, and
        conversely, every combinatorial model category arises in this way. We will also
        see that these constructions preserve right properness and compatibility with
        simplicial enrichment. Along the way, we establish some technical results on
        the index of accessibility of various constructions on accessible categories,
        which may be of independent interest.Comment: 44 pages, LaTeX. v4: Extended version of final journal version. (Note
          that material has been significantly reorganised since v3.",
              "downloadUrl": "http://arxiv.org/abs/1402.6659",
              "fullText": "ar
        X
        iv
        :1
        40
        2.
        66
        59
        v1
          [
        ma
        th.
        CT
        ]  
        26
         Fe
        b 2
        01
        4
        The heart of a combinatorial model category
        Zhen Lin Low
        27 February 2014
        Abstract
        We show that small model categories satisfying certain size condi-
        tions can be completed to yield a combinatorial model category, and
        conversely, that every combinatorial model category arises in this way.
        We also show that these constructions preserve right properness and
        compatibility with simplicial enrichment.
        Introduction
        Category-theoretic homotopy theory has seen a boom in recent decades. One
        development was the introduction of the notion of ‘combinatorial model cat-
        egories’ by Smith [1998]. These correspond to what Lurie [HTT] calls ‘present-
        able ∞-categories’ and are therefore a homotopy-theoretic generalisation of
        the locally presentable categories of Gabriel and Ulmer [1971]. The classiﬁc-
        ation of locally κ-presentable categories says that each one is equivalent to
        the free κ-ind-completion of a κ-cocomplete small category, and Lurie proved
        the analogous proposition for presentable ∞-categories, so it should at least
        seem plausible that every combinatorial model category is generated by a small
        model category in an appropriate sense.
        Indeed, the work of Beke [2000] suggests that more should be true. As
        stated in the abstract of the cited paper,
        If a Quillen model category can be speciﬁed using a certain logical
        syntax (intuitively, ‘is algebraic/combinatorial enough’), so that it
        can be deﬁned in any category of sheaves, then the satisfaction of
        Quillen’s axioms over any site is a purely formal consequence of
        their being satisﬁed over the category of sets.
        Department of Pure Mathematics and Mathematical Statistics, University of Cambridge,
        Cambridge, UK. E-mail address: Z.L.Low@dpmms.cam.ac.uk
        1
        The heart of a combinatorial model category
        and similarly, we can show that the question of whether a set of generating
        coﬁbrations and trivial coﬁbrations in a locally presentable category really do
        generate a combinatorial model category depends only on an essentially small
        full subcategory of small objects. More precisely:
        Theorem. Let M be a locally presentable category, and let I and I ′ be sub-
        sets of mor C. Suppose κ and λ are regular cardinals satisfying the following
        hypotheses:
        • M is a locally κ-presentable category, and κ is sharply less than λ.
        • Kλ(M), the full subcategory of M spanned by the λ-compact objects, is
        closed under finite limits in M.
        • There are < λ morphisms between any two κ-compact objects in M.
        • I and I ′ are λ-small sets of morphisms between κ-compact objects.
        Then the weak factorisation systems cofibrantly generated by I and I ′ underlie
        a model structure on M if and only if their restrictions to Kλ(M) underlie a
        model structure on Kλ(M).
        The techniques used in the proof of the above theorem are easily generalised,
        allowing us to make sense of a remark of Dugger [2001]:
        [. . . ] for a combinatorial category the interesting part of the ho-
        motopy theory is all concentrated within some small subcategory—
        beyond suﬃciently large cardinals the homotopy theory is somehow
        “formal”.
        For illustration, we will see how to validate Dugger’s principle in the case of
        right properness and axiom SM7.
        The author’s own motivation for seeking such a result is rooted in a more
        foundational question.",
              "fulltextIdentifier": "http://arxiv.org/abs/1402.6659",
              "fulltextUrls": Array [
                "https://core.ac.uk/display/25006269",
                "http://arxiv.org/abs/1402.6659",
              ],
              "id": "25006269",
              "identifiers": Array [
                "oai:arXiv.org:1402.6659",
                null,
              ],
              "language": Object {
                "code": "en",
                "id": 9,
                "name": "English",
              },
              "oai": "oai:arXiv.org:1402.6659",
              "relations": Array [],
              "repositories": Array [
                Object {
                  "description": null,
                  "disabled": false,
                  "id": "144",
                  "journal": null,
                  "lastUpdateTime": null,
                  "metadataFormat": null,
                  "name": "arXiv.org e-Print Archive",
                  "nrUpdates": 0,
                  "openDoarId": 0,
                  "pdfStatus": null,
                  "physicalName": "noname",
                  "repositoryLocation": null,
                  "roarId": 0,
                  "software": null,
                  "source": null,
                  "uri": null,
                  "uriJournals": null,
                  "urlHomepage": null,
                  "urlOaipmh": null,
                },
              ],
              "subjects": Array [
                "text",
              ],
              "title": "The heart of a combinatorial model category",
              "topics": Array [
                "Mathematics - Category Theory",
                "Mathematics - Algebraic Topology",
                "18G55, 55U35 (Primary), 18D35, 55P60 (Secondary)",
              ],
              "types": Array [],
              "year": 2016,
            },
          ]
    }
    isUserSignedIn={false}
    userAuthentication={
        Array [
            "9f9sv90b9z",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <article>
          <h1>
                    The heart of a combinatorial model category
          </h1>
          <h3>
                    Low, Zhen Lin
          </h3>
          <p>
                    We show that every small model category that satisfies certain size
                    conditions can be completed to yield a combinatorial model category, and
                    conversely, every combinatorial model category arises in this way. We will also
                    see that these constructions preserve right properness and compatibility with
                    simplicial enrichment. Along the way, we establish some technical results on
                    the index of accessibility of various constructions on accessible categories,
                    which may be of independent interest.Comment: 44 pages, LaTeX. v4: Extended version of final journal version. (Note
                      that material has been significantly reorganised since v3.
          </p>
</article>,
      ],
      "className": "user-dashboard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              The heart of a combinatorial model category
</h1>,
            <h3>
              Low, Zhen Lin
</h3>,
            <p>
              We show that every small model category that satisfies certain size
              conditions can be completed to yield a combinatorial model category, and
              conversely, every combinatorial model category arises in this way. We will also
              see that these constructions preserve right properness and compatibility with
              simplicial enrichment. Along the way, we establish some technical results on
              the index of accessibility of various constructions on accessible categories,
              which may be of independent interest.Comment: 44 pages, LaTeX. v4: Extended version of final journal version. (Note
                that material has been significantly reorganised since v3.
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "The heart of a combinatorial model category",
            },
            "ref": null,
            "rendered": "The heart of a combinatorial model category",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Low, Zhen Lin",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Low, Zhen Lin",
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "We show that every small model category that satisfies certain size
conditions can be completed to yield a combinatorial model category, and
conversely, every combinatorial model category arises in this way. We will also
see that these constructions preserve right properness and compatibility with
simplicial enrichment. Along the way, we establish some technical results on
the index of accessibility of various constructions on accessible categories,
which may be of independent interest.Comment: 44 pages, LaTeX. v4: Extended version of final journal version. (Note
  that material has been significantly reorganised since v3.",
            },
            "ref": null,
            "rendered": "We show that every small model category that satisfies certain size
conditions can be completed to yield a combinatorial model category, and
conversely, every combinatorial model category arises in this way. We will also
see that these constructions preserve right properness and compatibility with
simplicial enrichment. Along the way, we establish some technical results on
the index of accessibility of various constructions on accessible categories,
which may be of independent interest.Comment: 44 pages, LaTeX. v4: Extended version of final journal version. (Note
  that material has been significantly reorganised since v3.",
            "type": "p",
          },
        ],
        "type": "article",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <article>
            <h1>
                        The heart of a combinatorial model category
            </h1>
            <h3>
                        Low, Zhen Lin
            </h3>
            <p>
                        We show that every small model category that satisfies certain size
                        conditions can be completed to yield a combinatorial model category, and
                        conversely, every combinatorial model category arises in this way. We will also
                        see that these constructions preserve right properness and compatibility with
                        simplicial enrichment. Along the way, we establish some technical results on
                        the index of accessibility of various constructions on accessible categories,
                        which may be of independent interest.Comment: 44 pages, LaTeX. v4: Extended version of final journal version. (Note
                          that material has been significantly reorganised since v3.
            </p>
</article>,
        ],
        "className": "user-dashboard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                The heart of a combinatorial model category
</h1>,
              <h3>
                Low, Zhen Lin
</h3>,
              <p>
                We show that every small model category that satisfies certain size
                conditions can be completed to yield a combinatorial model category, and
                conversely, every combinatorial model category arises in this way. We will also
                see that these constructions preserve right properness and compatibility with
                simplicial enrichment. Along the way, we establish some technical results on
                the index of accessibility of various constructions on accessible categories,
                which may be of independent interest.Comment: 44 pages, LaTeX. v4: Extended version of final journal version. (Note
                  that material has been significantly reorganised since v3.
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "The heart of a combinatorial model category",
              },
              "ref": null,
              "rendered": "The heart of a combinatorial model category",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Low, Zhen Lin",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Low, Zhen Lin",
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "We show that every small model category that satisfies certain size
conditions can be completed to yield a combinatorial model category, and
conversely, every combinatorial model category arises in this way. We will also
see that these constructions preserve right properness and compatibility with
simplicial enrichment. Along the way, we establish some technical results on
the index of accessibility of various constructions on accessible categories,
which may be of independent interest.Comment: 44 pages, LaTeX. v4: Extended version of final journal version. (Note
  that material has been significantly reorganised since v3.",
              },
              "ref": null,
              "rendered": "We show that every small model category that satisfies certain size
conditions can be completed to yield a combinatorial model category, and
conversely, every combinatorial model category arises in this way. We will also
see that these constructions preserve right properness and compatibility with
simplicial enrichment. Along the way, we establish some technical results on
the index of accessibility of various constructions on accessible categories,
which may be of independent interest.Comment: 44 pages, LaTeX. v4: Extended version of final journal version. (Note
  that material has been significantly reorganised since v3.",
              "type": "p",
            },
          ],
          "type": "article",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
